pkg/app/server/httpapi/callback.go

```diff
@@ -148,8 +153,35 @@ func getUser(ctx context.Context, sso *model.ProjectSSOConfig, project *model.Pr
 			return nil, err
 		}
 		return cli.GetUser(ctx)
-
+	case model.ProjectSSOConfig_OIDC:
+		if sso.Oidc == nil {
+			return nil, fmt.Errorf("missing OIDC oauth in the SSO configuration")
+		}
+		cli, err := oidc.NewOAuthClient(ctx, sso.Oidc, project, code)
+		if err != nil {
+			return nil, err
+		}
+		return cli.GetUser(ctx, sso.Oidc.ClientId)
 	default:
 		return nil, fmt.Errorf("not implemented")
 	}
 }
+
+func parseProjectAndState(r *http.Request) (string, string, error) {
```

Comment:
Would you add unit tests of `parseProjectAndState()` in `callback_test.go`? (Sorry, the file is empty now...)